--- internal/common/expand_into_runfiles.bzl
+++ internal/common/expand_into_runfiles.bzl
@@ -32,22 +32,28 @@ def expand_location_into_runfiles(ctx, path):
     return expand_path_into_runfiles(ctx, path)

 def expand_path_into_runfiles(ctx, path):
-    """Expand a path into runfiles.
+    """Expand paths into runfiles.

-    Given a file path that might contain a $(location) label expansion,
-    provide the path to the file in runfiles.
+    Given a file path that might contain a $(location) or $(locations) label expansion,
+    provide the paths to the file in runfiles.

     See https://docs.bazel.build/versions/master/skylark/lib/ctx.html#expand_location

     Args:
       ctx: context
-      path: the path to expand
+      path: the paths to expand

     Returns:
-      The expanded path
+      The expanded paths
     """
     targets = ctx.attr.data if hasattr(ctx.attr, "data") else []
     expanded = ctx.expand_location(path, targets)
+
+    expansion = [resolve_expanded_path(ctx, exp) for exp in expanded.strip().split(" ")]
+
+    return " ".join(expansion)
+
+def resolve_expanded_path(ctx, expanded):
     if expanded.startswith("../"):
         return expanded[len("../"):]
     if expanded.startswith(ctx.bin_dir.path):